hack/pkg/merge_schemas.go

```diff
@@ -38,13 +39,21 @@ func RunMergeSchemas(valuesSchemaFile, platformConfigSchemaFile, outFile string)
 }
 
 func addPlatformSchema(platformSchema, toSchema *jsonschema.Schema) error {
+	collision := make(map[string]bool)
 	for defName, node := range platformSchema.Definitions {
 		if _, exists := toSchema.Definitions[defName]; exists {
-			panic("trying to overwrite definition " + defName + " this is unexpected")
+			collision[defName] = true
+			fmt.Printf("detected name collision: vCluster schema has %s definition, but platform schema has it too. Renaming platform one to %s\n", defName, "Platform"+defName)
+			toSchema.Definitions["Platform"+defName] = node
+			continue
 		}
 		toSchema.Definitions[defName] = node
 	}
 
+	for k, _ := range collision {
+		replaceRefInSchema(platformSchema, "#/$defs/"+k, "#/$defs/Platform"+k)
```

Comment:
Just curious, does this need to collect all the collisions, then do the replacement, or would it work to just call this in the loop above when you're building the collisions map?  Something like:

```go
	for defName, node := range platformSchema.Definitions {
		if _, exists := toSchema.Definitions[defName]; exists {
			fmt.Printf("detected name collision: vCluster schema has %s definition, but platform schema has it too. Renaming platform one to %s\n", defName, "Platform"+defName)
			toSchema.Definitions["Platform"+defName] = node
			replaceRefInSchema(platformSchema, "#/$defs/"+k, "#/$defs/Platform"+k)
			continue
		}
		toSchema.Definitions[defName] = node
	}
```

Reply:
good point, changed :+1: